cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_test.go

```diff
@@ -185,14 +233,216 @@ func TestGetAvailablePackages(t *testing.T) {
 				t.Fatalf("%+v", err)
 			}
 
-			if got, want := response.Packages, tc.expectedPackages; !cmp.Equal(got, want, cmp.Comparer(pkgEqual)) {
-				t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got, cmp.Comparer(pkgEqual)))
+			if got, want := response.Packages, tc.expectedPackages; !cmp.Equal(got, want, cmpopts.IgnoreUnexported(corev1.AvailablePackage{})) {
```

Comment:
I guess this is for ignoring the optional fields and just checking if the required fields are present, no?

Reply:
No, it's because `cmp.Equal` (and `cmp.Diff`) will by default error if they can't compare an unexported field, like this:
```
    --- FAIL: TestGetAvailablePackages/it_returns_carvel_packages_from_the_cluster (0.00s)
panic: cannot handle unexported field at {[]*v1alpha1.AvailablePackage}[1].state:
        "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1".AvailablePackage
consider using a custom Comparer; if you control the implementation of type, you can also consider using an Exporter, AllowUnexported, or cmpopts.IgnoreUnexported [recovered]
        panic: cannot handle unexported field at {[]*v1alpha1.AvailablePackage}[1].state:
        "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1".AvailablePackage
consider using a custom Comparer; if you control the implementation of type, you can also consider using an Exporter, AllowUnexported, or cmpopts.IgnoreUnexported

```